fatemeh/JS/JS: data fetching + DOM manipulation/index.html

```diff
@@ -0,0 +1,191 @@
+        let deletePost = confirm("Are you sure?");
+        if (deletePost) {
+          document
+            .getElementsByClassName(id)[0]
+            .getElementsByTagName("p")[0].style.color = "#384fc2";
+
+          document
+            .getElementsByClassName(id)[0]
+            .getElementsByTagName("p")[0].style.fontWeight = "bold";
+
+          document
+            .getElementsByClassName(id)[0]
+            .getElementsByTagName("p")[0].innerHTML = "Pending...";
+          fetch("https://jsonplaceholder.typicode.com/posts/".concat(id + 1), {
+            method: "DELETE",
+          })
+            .then((response) => response.json())
+
+            .then((json) =>
+              document
+                .getElementsByClassName("posts")[0]
+                .removeChild(document.getElementsByClassName(id)[0])
+            );
+        }
+      }
+      function addNewPost(id) {
+        location.replace("./addpost.html");
+      }
+      function cancelSearch() {
+        document.getElementsByClassName("SearchBox")[0].value = "";
+        fetchData();
+      }
+      async function mySearch(event) {
+        event.preventDefault();
+        let searchText = document.forms["Search"]["searchTerm"].value;
+        if (searchText == "") {
+        } else {
+          const response = await fetch(
+            `https://jsonplaceholder.typicode.com/posts?title=${searchText.replace(
+              /['"]+/g,
+              ""
+            )}`
+          );
+          const data = await response.json();
+          dataArray = data;
+          document.getElementsByClassName("posts")[0].innerHTML = "";
+          console.log(data);
+          for (let i = 0; i < data.length; i++)
+            document.getElementsByClassName(
+              "posts"
+            )[0].innerHTML += `<div class=${i} id=child>
+                        <h1>${dataArray[i]["title"]}</h1>
+                        <h3>Author: ${dataArray[i]["userId"]}</h3>
+                        <p>${dataArray[i]["body"]}</p>
+                        <div>
+                            <button class=update${i} id=update>Update</button>
+                            <button class=delete${i} id=delete>Delete</button>
+                        </div>
+                        </div>`;
+
+          for (let i = 0; i < data.length; i++) {
+            const updateButton = document.querySelector(`.update${i}`);
+            updateButton.addEventListener("click", () => {
+              Update(i);
+            });
+            const deleteButton = document.querySelector(`.delete${i}`);
+            deleteButton.addEventListener("click", () => {
+              Delete(i);
+            });
+          }
+          document.getElementsByClassName("Search")[0].innerHTML =
+            '<input class="SearchBox" type="text" name="searchTerm"placeholder=" Type here to search..." /><button type="submit" class="SearchButton">Search</button><button class="cancelButton">Cancel</button>';
```

Comment:
instead of replacing the entire content of the form, you can show or hide the cancel button based on whether the user is searching or not